core/merge.js

```diff
@@ -358,7 +358,11 @@ class Merge {
 
   // Move a folder and all the things inside it
   async moveFolderRecursivelyAsync (side /*: SideName */, folder /*: Metadata */, was /*: Metadata */, newRemoteRevs /*: ?RemoteRevisionsByID */) {
+    log.debug({path: folder.path, oldpath: was.path}, 'moveFolderRecursivelyAsync')
     const docs = await this.pouch.byRecursivePathAsync(was._id)
+
+    if (was.moveFrom && was.moveFrom.childMove) was.needNewRev = true
```

Comment:
Hmm... The fact that you feel the need to depend `moveFrom.childMove` + introduce a new `Metadata` property tells us that we (not you @taratatach :wink:) didn't make the right choice when introducing `#childMove` IMHO.

Reply:
That's one of the questions I asked @aenario when we paired on this one. I'm not sure I completely understand the need for the child move and its gain compared to adding a bunch of moves to the list of changes.